docs/topics/power-assert.md

```diff
@@ -0,0 +1,400 @@
+
+```kotlin
+fun <R> assertSoftly(block: AssertScope.() -> R): R {
+    val scope = AssertScopeImpl()
+    val result = scope.block()
+    if (scope.errors.isNotEmpty()) {
+        throw AssertionError(scope.errors.joinToString("\n"))
+    }
+    return result
+}
+
+interface AssertScope {
+    fun assert(assertion: Boolean, message: (() -> String)? = null)
+}
+
+class AssertScopeImpl : AssertScope {
+    val errors = mutableListOf<String>()
+    override fun assert(assertion: Boolean, message: (() -> String)?) {
+        if (!assertion) {
+            errors.add(message?.invoke() ?: "Assertion failed")
+        }
+    }
+}
+```
+
+Add these functions to the `powerAssert {}` block to make them available for the Power-assert plugin:
+
+```kotlin
+@OptIn(ExperimentalKotlinGradlePluginApi::class)
+powerAssert {
+    functions = listOf("kotlin.assert", "kotlin.test.assert", "org.example.AssertScope.assert")
+}
+```
+
+> You should specify the full name of the package where you declare the `AssertScope.assert()` function.
+>
+{type="tip"}
+
+After that, you could use it in your test code:
+
+```kotlin
+// Import the assertSoftly() function
+import org.example.assertSoftly
+        
+class SoftAssertExampleTest1 {
+
+    data class Employee(val name: String, val age: Int, val salary: Int)
+
+    @Test
+    fun `test employees data`() {
+        val employees = listOf(
+            Employee("Alice", 30, 60000),
+            Employee("Bob", 45, 80000),
+            Employee("Charlie", 55, 40000),
+            Employee("Dave", 150, 70000)
+        )
+
+        assertSoftly {
+            for (employee in employees) {
+                assert(employee.age < 100) { "${employee.name} has an invalid age: ${employee.age}" }
+                assert(employee.salary > 50000) { "${employee.name} has an invalid salary: ${employee.salary}" }
+            }
+        }
+    }
+}
+```
+
+In the output, all the `assert()` function error messages will be printed one after another:
+
+```text
+harlie has an invalid salary: 40000
+assert(employee.salary > 50000) { "${employee.name} has an invalid salary: ${employee.salary}" }
+       |        |      |
+       |        |      false
+       |        40000
+       Employee(name=Charlie, age=55, salary=40000)
+Dave has an invalid age: 150
+assert(employee.age < 100) { "${employee.name} has an invalid age: ${employee.age}" }
+       |        |   |
+       |        |   false
+       |        150
+       Employee(name=Dave, age=150, salary=70000)
+```
+
+## What's next
+
+* Look through a [simple project with the plugin enabled](https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-integration-tests/src/test/resources/testProject/powerAssertSimple)
+   and more [complex project with multiple source sets](https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-integration-tests/src/test/resources/testProject/powerAssertSourceSets)
```

Comment:
```suggestion
   and a more [complex project with multiple source sets](https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-integration-tests/src/test/resources/testProject/powerAssertSourceSets).
```

Should this be a list, with only a single item?

Reply:
Also, I think this link is wrong. I think `blob` needs to be `tree`. Unless you mean to link to the `build.gradle.kts` file?